### PR TITLE
Fix NPE if child channel has no parent (bsc#1231053)

### PR DESCRIPTION
**Pull request by mbussolotto**
## What does this PR change?

Fix NPE if child channel has no parent (bsc#1231053)

## GUI diff

No difference.
- [ ] **DONE**

## Documentation
- No documentation needed: **add explanation. This can't be used if there is a GUI diff**
- No documentation needed: only internal and user invisible changes
- Documentation issue was created: [Link for SUSE Manager contributors](https://github.com/SUSE/spacewalk/issues/new?template=ISSUE_TEMPLATE_DOCUMENTATION.md&labels=documentation&projects=SUSE/spacewalk/31), [Link for community contributors](https://github.com/uyuni-project/uyuni-docs/issues/new).
- API documentation added: please review the Wiki page [Writing Documentation for the API](https://github.com/uyuni-project/uyuni/wiki/Writing-documentation-for-the-API) if you have any changes to API documentation.
- (OPTIONAL) [Documentation PR](https://github.com/uyuni-project/uyuni-docs/pulls)

- [ ] **DONE**

## Test coverage
ℹ️ If a major new functionality is added, it is **strongly recommended** that tests for the new functionality are added to the Cucumber test suite
- No tests: **add explanation**
- No tests: already covered
- Unit tests were added
- Cucumber tests were added

- [ ] **DONE**

## Links

Issue(s): https://github.com/SUSE/spacewalk/issues/25417

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle" 
- [ ] Re-run test "java_pgsql_tests" 
- [x] Re-run test "ruby_rubocop"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

**Review by cbosdo — APPROVED** [review]
LGTM, but acceptance tests don't agree

File: java/code/src/com/redhat/rhn/frontend/action/common/DownloadFile.java [hunk 1]
```diff
@@ -282,7 +282,8 @@ else if (ksession != null) {
 
         if (map.containsKey(CHILD) && !Config.get().getBoolean("ks_restrict_child_channels")) {
             Channel child = ChannelFactory.lookupByLabel(map.get(CHILD));
-            if (child == null || tree == null || !child.getParentChannel().equals(tree.getChannel())) {
+            if (child == null || tree == null || child.getParentChannel() == null ||
+                    !child.getParentChannel().equals(tree.getChannel())) {
```
Comment (rjmateus) [inline java/code/src/com/redhat/rhn/frontend/action/common/DownloadFile.java:286]: sorry but how can a child channel have not parent? If it doesn't have a parent channel then it's not a child :D

Comment (mbussolotto) [reply java/code/src/com/redhat/rhn/frontend/action/common/DownloadFile.java:286]: looks like some reeeally strange  (and not supported...and pro)  setup might allow it. basically autoyast (that is supposed to be used just for child channels), was used with a parent channel: so  the child here was actually a parent. 

Besides the hacky way to reach this point, the code should prevents any NPE

Comment (rjmateus) [reply java/code/src/com/redhat/rhn/frontend/action/common/DownloadFile.java:286]: >the code should prevents any NPE

Sure, I was just surprise to see a child channel without a parent :)

**Review by rjmateus — APPROVED** [review]

**Pull request merged**